examples/ping-onion.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 Hannes Furmans
```

Comment:
I would suggest moving this example to `transports/onion/examples`. Not a strong opinion though.

Either way, I think it should be listed on `examples/README.md`.

Reply:
I think we should leave it there, since all examples are in the examples/ directory and not in the corresponding parts. If we change that for all, we should change it for libp2p-onion too, however I think this is more concise.

Reply:
> I think we should leave it there, since all examples are in the examples/ directory and not in the corresponding parts. If we change that for all, we should change it for libp2p-onion too, however I think this is more concise.

IMO the examples in `/examples` should showcase use of the root `libp2p` crate which usually involves multiple different protocols (identify + kad + ping etc). Most protocols in `/protocols` have their own examples that showcase specific aspects of them.

With this reasoning, the `ping` example shouldn't be there though and neither should `mdns-passive-discovery` and `gossipsub-chat`.